code/game/objects/items/puzzle_pieces.dm

```diff
@@ -86,6 +86,18 @@
 	puzzle_id = null //honestly these cant be closed anyway and im not fucking around with door code anymore
 	INVOKE_ASYNC(src, PROC_REF(try_puzzle_open), null)
 
+/obj/machinery/door/puzzle/animation_length(animation)
+	switch(animation)
+		if(DOOR_OPENING_ANIMATION)
+			return 1.0 SECONDS
+
+/obj/machinery/door/puzzle/animation_segment_delay(animation)
+	switch(animation)
+		if("opening_passable")
+			return 0.8 SECONDS
+		if("opening_done")
+			return 1.0 SECONDS
```

Comment:
Missed one, also `"opening_done"` seems unused?

Reply:
renamed to the finished suffix